cc/main.cc

```diff
@@ -456,13 +465,21 @@ class SelfPlayer {
       gflags::SetCommandLineOption(flag_value.first.c_str(),
                                    flag_value.second.c_str());
     }
+
+    if (FLAGS_resign_threshold < old_resign && old_resign == 1.0) {
+      // NOTE: This will trigger an interupt (and reload) of all current games.
+      required_flags_ = flags_timestamp_
+    }
   }
 
   absl::Mutex mutex_;
   std::unique_ptr<DualNetFactory> dual_net_factory_ GUARDED_BY(&mutex_);
   Random rnd_ GUARDED_BY(&mutex_);
   std::vector<std::thread> threads_;
+  // Timestamp of newest flag file.
   uint64_t flags_timestamp_ = 0;
```

Comment:
Since `flags_timestamp_` and `required_flags_` are compared from different threads, they either need to be protected by a mutex or change them to `std::atomic<uint64_t>`, which is probably simpler.

Reply:
writing is only done in MaybeReloadFlags which is guarded by mutex I believe this is sufficient

Reply:
Sadly not.

Since reading isn't guarded by a mutex, it's possible on 32bit architectures that a read could see a half-written value for `flags_timestamp_`.